File: internal/common/models/ticket.go
package models

type Ticket struct {
	ID      string  `json:"id"`
	Title   string  `json:"title"`
	Price   int64   `json:"price"`
	UserID  string  `json:"userId"`
	OrderID *string `json:"orderId,omitempty"`
	Version int     `json:"version"`
}

File: frontend/src/lib/types/ticket.ts
export interface Ticket {
  id: string;
  title: string;
  price: number;
  userId: string;
  version: number;
  orderId?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTicketInput {
  title: string;
  price: number;
}

export interface UpdateTicketInput {
  title?: string;
  price?: number;
}
